src/pathPlanning/LinePolygonsCrossings.cpp

```diff
@@ -144,29 +144,59 @@ void LinePolygonsCrossings::getBasicCombingPath(CombPath& combPath)
 
 void LinePolygonsCrossings::getBasicCombingPath(PolyCrossings& polyCrossings, CombPath& combPath) 
 {
+    // minimise the path length by measuring the length of both paths around the polygon so we can determine the shorter path
+
     ConstPolygonRef poly = boundary[polyCrossings.poly_idx];
     combPath.push_back(transformation_matrix.unapply(Point(polyCrossings.min.x - std::abs(dist_to_move_boundary_point_outside), transformed_startPoint.Y)));
-    if ( ( polyCrossings.max.point_idx - polyCrossings.min.point_idx + (int)poly.size() ) % poly.size()
-        < poly.size() / 2 )
-    { // follow the path in the same direction as the winding order of the boundary polygon
-        for(unsigned int point_idx = polyCrossings.min.point_idx
-            ; point_idx != polyCrossings.max.point_idx
-            ; point_idx = (point_idx < poly.size() - 1) ? (point_idx + 1) : (0))
-        {
-            combPath.push_back(PolygonUtils::getBoundaryPointWithOffset(poly, point_idx, dist_to_move_boundary_point_outside));
-        }
+
+    // follow the path in the same direction as the winding order of the boundary polygon
+    std::vector<Point> fwd_points;
+    Point prev = combPath.back();
+    coord_t fwd_len = 0;
+    for (unsigned int point_idx = polyCrossings.min.point_idx; point_idx != polyCrossings.max.point_idx; point_idx = (point_idx < poly.size() - 1) ? (point_idx + 1) : (0))
```

Comment:
I know the for loop works, can you make multiple lines of instructions ...( as was before) or simplify
 the "point_idx = (point_idx < poly.size() - 1) ? (point_idx + 1) : (0)"

Reply:
Really? OK, if it will make you happy.